include/picongpu/fields/MaxwellSolver/Lehe/Curl.hpp

```diff
@@ -123,56 +123,70 @@ namespace lehe
                 - float_X( 2.0 ) * beta_zy * isDir_z
                 - float_X( 3.0 ) * delta_dir0 * isDir_z;
 
+            /* Note: the rest of this function is made by copying and
+             * adapting the implementation for Y, including the sign reversals in
+             * indices, and reversals of terms for finite-difference derivatives
+             */
+
+            // Typedef an accessor to access mem[z][y][x]
+            // in (x,y,z) order :)
+            typedef DataSpace<DIM3> Space;
 
+            // actually computes -curl_x(E) = (-dEz/dy) - (-dEy/dz)
             const float_X curl_x
                 = (
-                    alpha_y * ( mem[0][0][0].z( ) - mem[0][-1][0].z( ) )
-                    + beta_yx * ( mem[1][0][0].z( ) - mem[1][-1][0].z( ) )
-                    + beta_yx * ( mem[-1][0][0].z( ) - mem[-1][-1][0].z( ) )
-                    ) * reci_dy
+                    alpha_y * ( mem(Space(0,0,0)*(-1)).z( ) - mem(Space(0,-1,0)*(-1)).z( ) )
```

Comment:
What does this double minus `- mem...` and `*(-1)` result in?

Reply:
I asked @psychocoderHPC and it seems no one remembers, this code is really old. We checked together with the paper and the implementation for Y (where these minuses came from) is correct, albeit weirdly formulated for an unknown reason.

Reply:
I think i will provide some refactoring, as this would also be fitting for the Lehe + PML solver. But as a separate PR, this is just a bug fix.